### PR TITLE
Bug 1880473: remove extraneous manifests in installplan 

### PR DESCRIPTION
**Pull request by exdx**
**Description of the change:**
Continues work from #1589 

**Motivation for the change:**
Regression

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

File: test/e2e/installplan_e2e_test.go [hunk 1]
```diff
@@ -2623,6 +2627,94 @@ var _ = Describe("Install Plan", func() {
...
+
+		// Ensure that all the steps have a configmap based reference
+		for _, step := range ip.Status.Plan {
+			manifest := step.Resource.Manifest
+			var ref catalog.UnpackedBundleReference
+			err := json.Unmarshal([]byte(manifest), &ref)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ref.Kind).To(Equal("ConfigMap"))
+		}
+	})
```
Comment (benluddy) [inline test/e2e/installplan_e2e_test.go:2717]: You might be able to streamline these assertions by [composing matchers](https://onsi.github.io/gomega/#composing-matchers) and using the [ConsistOf](http://onsi.github.io/gomega/#consistofelement-interface) matcher.

File: test/e2e/installplan_e2e_test.go [hunk 2]
```diff
@@ -2623,6 +2627,94 @@ var _ = Describe("Install Plan", func() {
...
+		createSubscriptionForCatalog(crc, catsrc.GetNamespace(), subName, catsrc.GetName(), "kiali", stableChannel, "", operatorsv1alpha1.ApprovalAutomatic)
+
+		sub, err := fetchSubscription(crc, catsrc.GetNamespace(), subName, subscriptionHasInstallPlanChecker)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Wait for the expected InstallPlan's execution to either fail or succeed
+		ipName := sub.Status.InstallPlanRef.Name
+		ip, err := waitForInstallPlan(crc, ipName, sub.GetNamespace(), buildInstallPlanPhaseCheckFunc(operatorsv1alpha1.InstallPlanPhaseFailed, operatorsv1alpha1.InstallPlanPhaseComplete))
+		Expect(err).ToNot(HaveOccurred())
+		Expect(operatorsv1alpha1.InstallPlanPhaseComplete).To(Equal(ip.Status.Phase), "InstallPlan not complete")
```
Comment (benluddy) [inline test/e2e/installplan_e2e_test.go:2682]: The expectation and actual values are swapped here, which doesn't matter to the test correctness, but does affect what is written to the test failure report.

**Review by benluddy — APPROVED** [review]
/approve

**Pull request merged**